src/nvim/terminal.c

```diff
@@ -295,7 +295,10 @@ void terminal_close(Terminal *term, char *msg)
   }
 
   if (buf) {
+    dict_T *dict = get_vim_var_dict(VV_EVENT);
+    tv_dict_add_nr(dict, "status", sizeof("status"), status);
```

Comment:
nit: could use S_LEN macro to replace these 2 params
```suggestion
    tv_dict_add_nr(dict, S_LEN("status"), status);
```